selectors/i18n.test.js

```diff
@@ -68,6 +68,92 @@ describe('selectors/i18n', () => {
 
             expect(getCurrentLocale(state)).toEqual(General.DEFAULT_LOCALE);
         });
+
+        describe('locale from query parameter', () => {
+            // Helper function to mock window.location.search with locale query parameter
+            const setWindowLocaleQueryParameter = (locale) => {
+                window.location.search = `?locale=${locale}`;
+            };
+
+            // Helper function to reset window.location.search
+            const resetWindowLocationSearch = () => {
+                window.location.search = '';
+            };
+
+            afterEach(() => {
+                resetWindowLocationSearch();
+            });
+
+            test('returns locale from query parameter if provided and not logged in', () => {
+                const state = {
+                    entities: {
+                        general: {
+                            config: {
+                                DefaultClientLocale: 'fr',
+                            },
+                        },
+                        users: {
+                            currentUserId: '',
+                            profiles: {},
+                        },
+                    },
+                };
+
+                setWindowLocaleQueryParameter('ko');
+
+                expect(getCurrentLocale(state)).toEqual('ko');
+
+                resetWindowLocationSearch();
```

Comment:
Since you call this in the `afterEach` above, I think we can skip it here so we're not repeating ourselves

Reply:
Forgot to clean this up, great catch! 😊